drivers/net/ethernet/intel/idpf/idpf_eth_idc.h

```diff
@@ -0,0 +1,199 @@
+ * @req_rel_vec_idx: Release vector indexes call to main driver
+ */
+struct idpf_eth_idc_ops {
+	void (*event_send)(struct idpf_eth_idc_dev_info *dev_info,
+			   struct idpf_eth_idc_event *event);
+	size_t (*virtchnl_send)(struct idpf_eth_idc_dev_info *dev_info,
+				struct idpf_vc_xn_params *params);
+	int (*intr_reg_init)(struct idpf_eth_idc_dev_info *dev_info,
+			     u16 num_vecs, struct idpf_q_vector *q_vectors,
+			      u16 *q_vector_idxs);
+	int (*intr_init_vec_idx)(struct idpf_eth_idc_dev_info *dev_info,
+				 u16 num_vecs,
+				 struct idpf_q_vector *q_vectors,
+				 u16 *q_vector_idxs);
+	int (*req_rel_vec_idx)(struct idpf_eth_idc_dev_info *dev_info,
+			       u16 *num_vectors,
+			       struct idpf_vector_info *vec_info,
+			       struct msix_entry *msix_table);
+};
+
+struct idpf_eth_idc_auxiliary_driver;
+
+/**
+ * struct idpf_eth_shared - Common Device data struct shared with eth
+ * @pdev: PCI device struct given on probe
+ * @msg_enable: Debug message level enabled
+ * @hw_addr: Hardware address for use by Ethernet driver
+ */
+struct idpf_eth_shared {
+	struct pci_dev *pdev;
+	struct idpf_eth_idc_ops eth_idc_ops;
+	u32 msg_enable;
+	void __iomem *hw_addr;
+};
+
+/**
+ * struct idpf_eth_idc_dev_info - Ethernet driver's device information struct
+ * @eth_shared: Ethernet shared data struct
+ * @vport_type: Vport type
+ * @caps: Auxiliary device capabilities
+ * @start_probe_post_init: Starts probe post initialization 
+ * @vport_id: Vport ID of device
+ * @idx: Index number of auxiliary device
+ */
+struct idpf_eth_idc_dev_info {
+	struct idpf_eth_shared *eth_shared;
+	enum idpf_vport_type vport_type;
+	struct idpf_eth_idc_auxiliary_dev_caps caps;
+	bool start_probe_post_init;
+	u32 vport_id;
+	u16 idx;
+};
+
+/**
+ * struct idpf_eth_idc_auxiliary_dev - Represents auxiliary device
+ * @adev: Auxiliary device
+ * @eth_info: Ethernet private data struct
+ */
+struct idpf_eth_idc_auxiliary_dev {
+	struct auxiliary_device adev;
+	struct idpf_eth_idc_dev_info eth_info;
+};
+
+/**
+ * struct idpf_eth_idc_auxiliary_driver - Ethernet driver info struct
+ * @signature: Signature value
+ * @adrv: Auxiliary driver
+ * @event_handler: Ethernet driver's event handler
+ */
+struct idpf_eth_idc_auxiliary_driver {
+	unsigned long signature;
+	struct auxiliary_driver adrv;
+	void (*event_handler)(struct idpf_eth_idc_dev_info *dev_info,
+			      struct idpf_eth_idc_event *event);
+};
+
+#define idpf_dev_info_to_adapter(dev_info) \
+	((struct idpf_adapter *)((dev_info)->idpf_context))
+#define idpf_eth_shared_to_dev(eth_shared) (&(eth_shared)->pdev->dev)
+
+int idpf_eth_idc_init_shared(struct idpf_eth_shared *eth_shared,
+			     struct pci_dev *pdev);
+void idpf_eth_idc_deinit_shared(struct idpf_eth_shared *eth_shared);
+
+void idpf_eth_idc_device_init(struct idpf_adapter *adapter);
+void idpf_eth_idc_device_deinit(struct idpf_adapter *adapter);
+
+int idpf_eth_idc_driver_register(void);
```

Comment:
i think this also belongs to pci driver..... so should be moved out of this header

Reply:
done !